### PR TITLE
[8.x] Adds a few PHP 8.1 related changes

### PR DESCRIPTION
**Pull request by driesvints**
This PR adds some additional fixes for PHP 8.1 support. Thanks to @GrahamCampbell for his help with this and third party library support.

### Todo

- [x] Check into `PDO::ATTR_STRINGIFY_FETCHES` behavior: https://github.com/laravel/framework/issues/36813#issuecomment-852418759
- [x] Mockery fix: https://github.com/mockery/mockery/pull/1144
- [ ] `predis/predis` PHP 8.1 compatibility
- [x] `ramsey/uuid`: https://github.com/ramsey/uuid/pull/388
- [ ] `aws/aws-php-sdk` serializable methods: https://github.com/aws/aws-sdk-php/pull/2316
- [ ] `aws/aws-php-sdk` `DateTimeResult::fromTimestamp` issue: https://github.com/laravel/framework/runs/3599702797#step:8:166
- [ ] Argon2 hashing issue: https://github.com/laravel/framework/runs/3599702797#step:8:126
- [ ] Fix all tests
- [x] Enable build failures

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
image

File: composer.json [hunk 1]
```diff
@@ -83,7 +83,7 @@
         "filp/whoops": "^2.8",
         "guzzlehttp/guzzle": "^6.5.5|^7.0.1",
         "league/flysystem-cached-adapter": "^1.0",
-        "mockery/mockery": "^1.4.2",
+        "mockery/mockery": "1.4.x-dev",
```
Comment (GrahamCampbell) [inline composer.json:86]: ^1.4.3

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (driesvints) [reply composer.json]: Great call!

**Review by kylekatarnls — APPROVED** [review]

**Comment by GrahamCampbell** [comment]
@kylekatarnls  why have you approved a PR in a broken state?

**Comment by kylekatarnls** [comment]
I would like to see `ReturnTypeWillChange` merged which should unlock the composer install for Laravel. And I think we could have 1 other PR first to add all the `#[ReturnTypeWillChange]` then the rest in a second step with conflicts fixed when ready.

**Comment by GrahamCampbell** [comment]
It’s not ready yet. It’ll probably be another few weeks, at the earliest. Even if the framework code is ready, it’s dependencies are not all ready, which means it still won’t work in practice.

**Comment by kylekatarnls** [comment]
I know, I just released a new fix today for Carbon compatibility with PHP 8.1 interfaces that added new return types.

**Comment by driesvints** [comment]
> I would like to see ReturnTypeWillChange merged which should unlock the composer install for Laravel

I can install Laravel v9 just fine right now. Is there anything custom you're doing maybe that's preventing you to do that?

**Comment by GrahamCampbell** [comment]
> Check into PDO::ATTR_STRINGIFY_FETCHES behavio

I think this is covered now. :)

**Review by GrahamCampbell — APPROVED** [review]

**Comment by nunomaduro** [comment]
@taylorotwell I've skipped the tests that are currently failing in PHP 8.1, so we can make this pull request in a mergeble state.

**Pull request merged**